.github/workflows/setup-java-maven.yml

```
@@ -0,0 +1,35 @@
+name: Java and Maven Setup
+
+on:
+  workflow_call:  # This workflow can be called from other workflows
+
+jobs:
+  setup-java-maven:
+    runs-on: self-hosted
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
```

actionlint (1.7.9)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)